Use fp64 result for correctness checking in TorchBench

desertfire opened the pull request:
Summary: Compare model outputs with the fp64 eager result. The idea is to use fp64 as the golden baseline and compare the difference between eager_model/fp64_model and optimized_model/fp64_model. This also makes the comparison consistent with the approach used by PT2.

Differential Revision: D42637398

xuzhao9 commented:
Does this PR fix https://github.com/pytorch/benchmark/issues/1354, or it is still unknown why there's inconsistency in the accuracy result?

desertfire commented:
> Does this PR fix #1354, or it is still unknown why there's inconsistency in the accuracy result on resnet50?

I just gave it a try, and the benchmark does pass now, although it hit `fp64 golden ref were not generated. Setting accuracy check to cosine`. This PR gets us one step closer to being consistent with the checking in PyTorch core. We can keep addressing those issues if there are more inconsistency coming up.

xuzhao9 commented:
> > Does this PR fix #1354, or it is still unknown why there's inconsistency in the accuracy result on resnet50?
> 
> I just gave it a try, and the benchmark does pass now, although it hit `fp64 golden ref were not generated. Setting accuracy check to cosine`. This PR gets us one step closer to being consistent with the checking in PyTorch core. We can keep addressing those issues if there are more inconsistency coming up.

Curious does this problem also happen in PyTorch core torchbench.py?

xuzhao9 commented:
https://github.com/pytorch/benchmark/pull/1700 should address this
